### DISCUSSION TITLE
Authenticate User Server-Side (nodejs + sockets) with token

### DISCUSSION THREAD

**Discussion by johnseabird** [pocketbase/pocketbase#3128] [Q&A] [answered]
Hey everyone, I'm thinking about using pocketbase in my nodejs / sockets app. There is no web front-end that could use the authStore in that case (it is a game application) since there is no SDK for it. Or am I getting something wrong here?

My idea was: users register and authenticate via the rest-like api of PocketBase and get a token back. I connect to my nodejs app with sockets. 
I would love to send the auth token the client received from the authentication to my NodeJS / Sockets app on connection to checks if the user is authenticated based on the token the client submits with the sockets custom header.

the server could validate via the rest-like api and send the token with an authentication header? I do not care for now. But how would I do that?

I'm a little confused on how I would authenticate / check if the token is valid on the server?

Thanks for reading.

**Comment by ganigeorgiev** [comment, marked as answer]
> I'm a little confused on how I would authenticate / check if the token is valid on the server?

We don't have an explicit action that will just verify whether a token is valid or not.
For such cases usually the [`authRefresh()` call is used](https://pocketbase.io/docs/api-records/#auth-refresh) - on valid auth token it will return the latest user data and a new token with refreshed `exp` claim (note that the last token is still valid); on invalid or expired token it returns an error.

**Reply by johnseabird** [reply]
Perfect, thank you very much. I'll give it a try. I missed the fact, that the old token is still valid after using authRefresh() while I was researching - so I do not have to update anything on the client side or inform the client about an updated token.
